tests/tools/agent_file.py

```diff
@@ -0,0 +1,78 @@
+import io
+import os
+import threading
+import time
+from pathlib import Path
+from typing import Optional, AnyStr, Generator
+
+import docker
+import docker.types
+
+
+class MountedFile(io.TextIOBase):
+    DEFAULT_READ_TIMEOUT = 10
+
+    def __init__(
+            self,
+            container_path: str,
+            host_path: Path,
+            name: Optional[str] = None,
+            type: str = "bind",
+            mode: str = "r",
+            content: Optional[AnyStr] = None
+    ):
+        self.container_path = container_path
+
+        if name is None:
+            name = container_path.replace(os.sep, "_")
+            host_path = host_path / name
+
+        self.host_path = host_path
+        self._type = type
+        self._mode = mode
+        self._initial_content = content
+
+        self._file_obj = None
+        self._file_obj_lock = threading.Lock()
+        self._buffer = io.StringIO()
+
+    def create(self):
+        if not self.host_path.exists():
+            self.host_path.touch()
+        if self._initial_content:
+            self.write_content(self._initial_content)
+
+        self._file_obj = self.host_path.open(self._mode)
+
+    @property
+    def docker_mount(self):
+        return docker.types.Mount(
+            str(self.container_path),
+            str(self.host_path),
+            type=self._type,
+            consistency="consistent"
+        )
+
+    def write_content(self, data: str):
+        self.host_path.write_text(data)
+
+    def read_content(self):
+        return self.host_path.read_text()
+
+    def read(self, size: Optional[int] = ...) -> str:
```

Comment:
We still support Python 2.x at the moment, so for the time being, we still need to utilize Python 2 compatible syntax for type annotations (aka type annotations inside comments).

But yeah, in the future once we drop support for Python 2, we can start to utilize native type annotation syntax which is much nicer.

Another option is we only run those tests under Python 3, but then it will still be a bit confusing / messy since different code may contain different type annotations notation.

Reply:
But overall, it's good that any new code contains type annotations :+1: